File: frontend/src/types/trades.ts
export interface TradeRow {
  id: number
  instrument_id: number
  symbol: string | null
  isin: string | null
  trade_date: string
  trade_type: string
  quantity: number
  price: number
  amount: number | null
  brokerage: number
  exchange: string | null
  segment: string | null
  notes: string | null
  source: string
  import_batch_id: string | null
  order_id: string | null
}

export interface TradeOrderRow {
  order_id: string
  instrument_id: number
  symbol: string | null
  isin: string | null
  trade_date: string
  trade_type: string
  quantity: number
  price: number
  amount: number
  exchange: string | null
  segment: string | null
  source: string
  trades: TradeRow[]
}

export interface TradesListResponse {
  rows: TradeOrderRow[]
  page: number
  per_page: number
  total: number
  total_pages: number
  q: string
}

export interface ImportFileResult {
  filename: string
  batch_id: string
  row_count: number
  success_count: number
  errors: Array<{ row: number; message: string }>
}

export interface ImportResponse {
  results: ImportFileResult[]
  holdings_count: number
  violations: Violation[]
}

export interface Violation {
  symbol?: string
  isin?: string
  kind: string
  detail?: string
}

export interface ImportBatch {
  id: number
  batch_id: string
  filename: string | null
  imported_at: string
  row_count: number | null
  success_count: number | null
  error_count: number | null
}

File: app/schemas/trades.py
from typing import Optional
from pydantic import BaseModel


class TradeRow(BaseModel):
    id: int
    instrument_id: int
    symbol: Optional[str] = None
    isin: Optional[str] = None
    trade_date: str
    trade_type: str
    quantity: float
    price: float
    amount: Optional[float] = None
    brokerage: float
    exchange: Optional[str] = None
    segment: Optional[str] = None
    notes: Optional[str] = None
    source: str
    import_batch_id: Optional[str] = None


class TradesListResponse(BaseModel):
    rows: list[TradeRow]
    page: int
    per_page: int
    total: int
    total_pages: int
    q: str


class ImportFileResult(BaseModel):
    filename: str
    batch_id: str
    row_count: int
    success_count: int
    errors: list[dict]


class Violation(BaseModel):
    symbol: Optional[str] = None
    isin: Optional[str] = None
    kind: str
    detail: Optional[str] = None


class ImportResponse(BaseModel):
    results: list[ImportFileResult]
    holdings_count: int
    violations: list[dict]


class ImportBatch(BaseModel):
    id: int
    batch_id: str
    filename: Optional[str] = None
    imported_at: str
    row_count: Optional[int] = None
    success_count: Optional[int] = None
    error_count: Optional[int] = None
